assets/js/src/core/components/tree/components/context-menu/asset-tree-context-menu.tsx

```diff
@@ -26,14 +41,116 @@ export interface AssetTreeContextMenuProps {
 
 export const AssetTreeContextMenu = (props: AssetTreeContextMenuProps): React.JSX.Element => {
   const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+  const { addJob } = useJobs()
   const { uploadFile: uploadFileProcessor, uploadZip: uploadZipProcessor } = UseFileUploader({ parentId: props.node?.id })
   const uploadFileRef = React.useRef<HTMLButtonElement>(null)
   const uploadZipRef = React.useRef<HTMLButtonElement>(null)
+  const [defaultValue, setDefaultValue] = useState<Store>({})
+  const [confirmationText, setConfirmationText] = useState<string | React.JSX.Element>('')
+  const [assetRename] = useAssetPatchByIdMutation()
+  const [assetDelete] = useElementDeleteMutation()
+  const {
+    addFolder,
+    rename,
+    cut,
+    copy,
+    paste,
+    remove,
+    downloadAsZip,
+    advanced,
+    refresh,
+    requestTranslations
+  } = useAssetActions()
+
+  const onRenameSubmit = async (form: FormInstance<any>): Promise<void> => {
+    const nodeId = parseInt(props.node!.id)
+    const assetRenameTask = assetRename({
+      body: {
+        data: [{
+          id: nodeId,
+          key: form.getFieldValue('input')
+        }]
+      }
+    })
+
+    try {
+      await assetRenameTask
+
+      dispatch(
+        assetApi.util.invalidateTags(
+          invalidatingTags.ASSET_TREE_ID(parseInt(props.node!.parentId!))
+        )
+      )
+    } catch (error) {
+      console.error('Error renaming asset', error)
+    }
+  }
+
+  const onConfirmation = async (): Promise<void> => {
+    const isFolder = props.node!.type === 'folder'
+
+    // TODO: add multiple deletion // this also requires mercure
+    if (isFolder) {
+      addJob(createDeleteJob({
+        title: 'Deleting Folder',
+        topics: [topics['deletion-finished'], ...defaultTopics],
+        action: async () => {
+          const promise = assetDelete({
+            id: parseInt(props.node!.id),
+            elementType: 'asset'
+          })
+
+          promise.catch(() => {
+            console.error('Error deleting asset')
+          })
+
+          const response = (await promise) as any
+
+          if (response.data === undefined || response.data === null) {
+            throw new Error(response.error.data.message as string ?? 'Error deleting Asset')
+          }
+
+          const data = response.data as { id: number }
+          return data.id
+        },
+        parentFolder: props.node!.parentId!
+      }))
+    }
+
+    try {
+      if (!isFolder) {
+        await assetDelete({
+          id: parseInt(props.node!.id),
+          elementType: 'asset'
+        })
+
+        dispatch(
+          assetApi.util.invalidateTags(
+            invalidatingTags.ASSET_TREE_ID(parseInt(props.node!.parentId!))
+          )
+        )
+      }
+    } catch (error) {
+      console.error('Error deleting asset', error)
+    }
+  }
+
+  const {
+    showModal: showInputModal,
+    renderModal: InputModal
+  } = useFormModal<InputModal>({ type: 'input' })
```

Comment:
Hm... I'm still not 100% sure if this is the best approach for our prompts. It is still quite a bit of work to add a simple prompt. Not sure if it's easily possible but in my dreams we would have a simple syntax similar to this without the need of adding so much of a custom code:

```javascript
prompt('Please enter the new name', (value) => console.log('entered value: ' + value)
```

```javascript
confirm('Do you really want to delete this item?', () => console.log('thanks for pressing ok, I will delete it for you!')
```

Maybe we need some additional parameters (or a config object) for the title etc. instead of only a simple message, but It would be much easier to use in my opinion. WDYT?